tests/test_execution.py

```diff
@@ -221,7 +226,8 @@ def test_function_build_failure(factory, network, default_dataset_1, worker):
 
         assert traintask.status == Status.failed
         assert traintask.error_type == substra.sdk.models.TaskErrorType.build
-        assert traintask.outputs[OutputIdentifiers.model].value is None
+        with pytest.raises(ValueError):
```

Comment:
I feel like the `ValueError` is less readable than the `is None` was. 
Would it be possible to get the outputs and check that the model is None or that the key is not present? 
Same for below.

Reply:
Thibault was suggesting creating errors (`AssetNotFoundError`) instead of `ValueError`. Would checking if it raises `AssetNotFoundError` clear for you?

Reply:
Yes, I think that would be much clearer